Duplicati/Library/AutoUpdater/UpdaterManager.cs

```diff
@@ -316,10 +318,16 @@ public static UpdateInfo CheckForUpdate(ReleaseType channel = ReleaseType.Unknow
 
                     using (var tmpfile = new Library.Utility.TempFile())
                     {
-                        System.Net.WebClient wc = new System.Net.WebClient();
-                        wc.Headers.Add(System.Net.HttpRequestHeader.UserAgent, string.Format("{0} v{1}{2}", APPNAME, SelfVersion.Version, string.IsNullOrWhiteSpace(InstallID) ? "" : " -" + InstallID));
-                        wc.Headers.Add("X-Install-ID", InstallID);
-                        wc.DownloadFile(url, tmpfile);
+
+                        using var request = new HttpRequestMessage(HttpMethod.Get, url);
+
+                        request.Headers.Add(System.Net.HttpRequestHeader.UserAgent.ToString(), string.Format("{0} v{1}{2}", APPNAME, SelfVersion.Version, string.IsNullOrWhiteSpace(InstallID) ? "" : " -" + InstallID));
+                        request.Headers.Add("X-Install-ID", InstallID);
+
+                        using var timeoutToken = new CancellationTokenSource();
+                        timeoutToken.CancelAfter(HttpContextSettings.ReadWriteTimeout);

```

Comment:
This should be `HttpContextSettings.OperationTimeout` as that is the overall operation timeout.
I think `HttpContextSettings.ReadWriteTimeout` will have to be deprecated as this concept is not supported with `HttpClient`.